tests/test_cat.py

```diff
@@ -388,6 +390,46 @@ def test_hashing(self):
         cat = CAT.load_model_pack(os.path.join(save_dir_path.name, f"{full_model_pack_name}.zip"))
         self.assertEqual(cat.get_hash(), cat.config.version.id)
 
+    def assert_patch_called(self, method, name: str, name_status: str, nr_of_calls: int):
+        cui = 'CUI-%d'%(hash(name) + id(name))
+        self.undertest.add_and_train_concept(cui, name, name_status=name_status)
+        if nr_of_calls == 0:
+            method.assert_not_called()
+        elif nr_of_calls == 1:
+            method.assert_called_once()
+        else:
+            raise ValueError("Undefined")
+
+    @unittest.mock.patch('medcat.cat.logger')
+    def test_add_and_train_concept_cat_nowarn_long_name(self, cat_logger):
```

Comment:
Just tried that. And it works fine for 3.10+, but fails in 3.8 and 3.9 since these do not have the `assertNoLogs` method yet. The method was added in 3.10.
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNoLogs

But I think I should be able to engineer a workaround.